addons/website_event_sale/models/sale_order.py

```diff
@@ -40,11 +40,36 @@ def _website_product_id_change(self, order_id, product_id, qty=0, **kwargs):
             values['product_id'] = ticket.product_id.id
             values['event_id'] = ticket.event_id.id
             values['event_ticket_id'] = ticket.id
+            discount = 0
+            ticket_currency = ticket.product_id.currency_id
+            pricelist_currency = order.pricelist_id.currency_id
+            price = ticket.price
+            price_reduce = ticket.price_reduce
+            if ticket_currency != pricelist_currency:
+                price = ticket_currency._convert(
+                    price,
+                    pricelist_currency,
+                    order.company_id,
+                    order.date_order or fields.Datetime.now()
+                )
+                price_reduce = ticket_currency._convert(
+                    price_reduce,
+                    pricelist_currency,
+                    order.company_id,
+                    order.date_order or fields.Datetime.now()
+                )
             if order.pricelist_id.discount_policy == 'without_discount':
-                values['price_unit'] = ticket.price
+                price_unit = price
+                discount = (price_unit - price_reduce) / price_unit * 100
```

Comment:
What if unit price is 0 ? you should make sure it won't crash bc of a division by 0